python/lsst/pipe/tasks/dcrAssembleCoadd.py

```diff
@@ -0,0 +1,781 @@
+# the GNU General Public License along with this program.  If not,
+# see <https://www.lsstcorp.org/LegalNotices/>.
+#
+
+from collections import namedtuple
+import numpy
+import scipy.ndimage.interpolation
+from lsst.afw.coord.refraction import differentialRefraction
+import lsst.afw.geom as afwGeom
+import lsst.afw.image as afwImage
+import lsst.afw.math as afwMath
+import lsst.coadd.utils as coaddUtils
+import lsst.pex.config as pexConfig
+import lsst.pipe.base as pipeBase
+from lsst.pipe.tasks.assembleCoadd import _subBBoxIter
+from lsst.pipe.tasks.assembleCoadd import AssembleCoaddTask
+from lsst.pipe.tasks.assembleCoadd import CompareWarpAssembleCoaddTask
+from lsst.pipe.tasks.assembleCoadd import CompareWarpAssembleCoaddConfig
+
+__all__ = ["DcrAssembleCoaddTask"]
+
+
+class DcrAssembleCoaddConfig(CompareWarpAssembleCoaddConfig):
+    filterName = pexConfig.Field(
+        dtype=str,
+        doc="Name of the band-defining filter of the observations.",
+        default='g',
+    )
+    bufferSize = pexConfig.Field(
+        dtype=int,
+        doc="Number of pixels to grow the subregion bounding box by.",
+        default=5,
+    )
+    useFFT = pexConfig.Field(
+        dtype=bool,
+        doc="Option to use use Fourier transforms for the convolutions.",
+        default=False,
+    )
+    usePsf = pexConfig.Field(
+        dtype=bool,
+        doc="Option to use the PSF as part of the convolution; requires `useFFT=True`.",
+        default=False,
+    )
+    dcrNSubbands = pexConfig.Field(
+        dtype=int,
+        doc="Number of sub-bands to forward model chromatic effects to fit the supplied exposures.",
+        default=3,
+    )
+    maxNIter = pexConfig.Field(
+        dtype=int,
+        doc="Maximum number of iterations of forward modeling.",
+        default=8,
+    )
+    minNIter = pexConfig.Field(
+        dtype=int,
+        doc="Minimum number of iterations of forward modeling.",
+        default=3,
+    )
+    convergenceThreshold = pexConfig.Field(
+        dtype=float,
+        doc="Target change in convergence between iteration of forward modeling.",
+        default=0.001,
+    )
+    useConvergence = pexConfig.Field(
+        dtype=bool,
+        doc="Turn on or off the convergence test as a forward modeling end condition.",
+        default=True,
+    )
+    doWeightGain = pexConfig.Field(
+        dtype=bool,
+        doc="Use the calculated convergence metric to accelerate forward modeling.",
+        default=True,
+    )
+    doAirmassWeight = pexConfig.Field(
+        dtype=bool,
+        doc="Weight exposures by airmass.",
+        default=True,
+    )
+    clampModel = pexConfig.Field(
+        dtype=float,
+        doc="Restrict new solutions from changing by more than a factor of `clampModel`.",
+        default=2.,
+    )
+    regularizeSigma = pexConfig.Field(
+        dtype=float,
+        doc="Set the threshold to exclude noise-like pixels from frequency regularization.",
```

Comment:
"Set" is a verb. You want a noun to describe what this field is. Maybe just drop "Set the"?